### PR TITLE
Build base image with maven and use it in the stress tests

### LINKED ISSUE #37346: [FEATURE REQ] Create a base stress Docker image for Java compilation.
Java has some build tooling that needs to be installed, packaging this up into a single base image. This is distinct from the runtime image which we need to run.

1. It needs to `mvn install` the required azure-sdk-tools and checkstyle plugins.
2. Probably more.

The only person that can handle this properly is @lmolkova, or if she's not available @alzimmermsft.

### PR DESCRIPTION
**Pull request by lmolkova**
Fix #37346

### CODE REVIEW

File: eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM mcr.microsoft.com/openjdk/jdk:11-mariner as builder
```
Comment (benbp) [inline eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile:1]: We can't check in files to `eng/common` as they are synced from azure-sdk-tools to all repos.

File: sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM maven:3.8.6-openjdk-11 as builder
+FROM stresspgs7b6dif73rup6.azurecr.io/java-tools/mvn:java11-mvn3.8.8 as builder
```
Comment (benbp) [inline sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile:1]: We should probably put this in the azsdkengsys acr? Or mirror it into the ACR used for the weekly prod deploys.

File: eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+RUN yum -y install curl tar
```
Comment (benbp) [inline eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile:3]: This should keep the image up to date on every build so we don't get OS vuln warnings later on:

```suggestion
RUN yum -y update
RUN yum -y install curl tar
```

File: sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile (hunk 2 above)
Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile:1]: We (non-eng folks) don't have push access to azsdkengsys (at least I don't, I do have pull access though).
So we'd need to kindly ask you and other eng friends to push it for us every once in a while. Would you do?

File: eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+ARG MAVEN_VERSION="3.8.8"
```
Comment (alzimmermsft) [inline eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile:5]: Any reason we don't use the latest Maven version?

File: sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile:1]: Agreed, these containers should be long-lived as we neither need to upgrade the Maven or JDK versions aggressively nor do either ship frequently (Java LTSes every two years, Maven releases roughly every few months).

File: eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile (hunk 1 above)
Comment (alzimmermsft) [inline eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile:1]: Will there be a template for each LTS version? Or could this be parameterized to handle all Java LTS versions (8, 11, 17, 21)

File: sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile (hunk 2 above)
Comment (benbp) [reply sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile:1]: @lmolkova I think we can follow a similar model here as we do for net. If you want to copy from https://github.com/Azure/azure-sdk-for-net/tree/main/eng/containers I can set up a pipeline with publish permissions and we can use that?

[the remark this answers is not in the document]
Comment (benbp) [reply sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile:1]: Yes see https://github.com/Azure/azure-sdk-for-java/pull/37350#discussion_r1372255804 I think we can start with that approach.

Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile]: Sounds awesome! I updated the PR to implement it.

[the remark this answers is not in the document]
Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus-stress/Dockerfile]: thanks, done

File: eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN mkdir -p /usr/share/maven \
+    && curl -Lso  /tmp/maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
+    && tar -xzC /usr/share/maven --strip-components=1 -f /tmp/maven.tar.gz \
+    && rm -v /tmp/maven.tar.gz \
+    && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
+
+ENV MAVEN_HOME /usr/share/maven
+
+# docker build -t <registry>.azurecr.io/java-tools/mvn:java11-mvn3.8.8
```
[the remark this answers is not in the document]
Comment (lmolkova) [reply eng/common/scripts/stress-testing/java11-mvn-mariner-dockerfile]: added doc and recommendation to also add `jdk{version}-latest` tag when updating maven

**Review by benbp — APPROVED** [review]

**Pull request merged**